.ci/test-python-oldest.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# oldest versions of dependencies published after
+# minimum supported Python version's first release
+#
+# see https://devguide.python.org/versions/
+#
+echo "installing lightgbm's dependencies"
+pip install \
+  'numpy==1.12.0' \
+  'pandas==0.24.0' \
+  'scikit-learn==0.18.2' \
+  'scipy==0.19.0' \
+|| exit -1
+echo "done installing lightgbm's dependencies"
```

Comment:
For example... `lightgbm` currently supports Python `>= 3.6`.

https://github.com/microsoft/LightGBM/blob/ac57d5a45a3ff3a04fbbcb8fb24f0c60072471e2/python-package/pyproject.toml#L26

According to https://devguide.python.org/versions/, the first release in that series was December 23, 2016.

So these are each the first release published to PyPI after December 23, 2016 (except `pandas==0.24.0`, which was later but the lowest version `lightgbm` support, see #5937).

Reply:
Why set these to their lowest versions? People using py36 will most likely be able to get the latest versions of these libraries released for that python version.

Reply:
Setting to the lowest versions makes it possible to automatically catch incompatibilities with older versions of these libraries, like those mentioned in #5937.

Having such things automatically brought to our attention during PR review means we can make informed decisions about whether to increase the floor on certain dependencies or to introduce compatibility patches like these:

https://github.com/microsoft/LightGBM/blob/9f78cceee4911dd56f4635dfd36d4482363db5aa/python-package/lightgbm/basic.py#L702-L711

https://github.com/microsoft/LightGBM/blob/9f78cceee4911dd56f4635dfd36d4482363db5aa/python-package/lightgbm/compat.py#L11-L14